src/main/java/org/ballerinax/datamapper/DataMapperStructureVisitor.java

```diff
@@ -105,25 +104,31 @@ public void visit(BLangSimpleVariable varNode) {
     public void visit(BLangRecordLiteral.BLangRecordKeyValueField recordKeyValue) {
         switch(recordKeyValue.valueExpr.getKind()) {
             case LITERAL:
-                ((BLangLiteral)recordKeyValue.valueExpr).accept(this);
+                ((BLangLiteral) recordKeyValue.valueExpr).accept(this);
                 break;
             case RECORD_LITERAL_EXPR:
             case LIST_CONSTRUCTOR_EXPR:
             case SIMPLE_VARIABLE_REF:
             case NUMERIC_LITERAL:
                 recordKeyValue.valueExpr.accept(this);
                 break;
+            default:
+                break;
         }
     }
 
     @Override
     public void visit(BLangRecordLiteral recordLiteral) {
-        for(RecordLiteralNode.RecordField field : recordLiteral.fields) {
+        for (RecordLiteralNode.RecordField field : recordLiteral.fields) {
             if (field.getKind() == NodeKind.RECORD_LITERAL_KEY_VALUE) {
-                ((BLangRecordLiteral.BLangRecordKeyValueField) field).accept(this);
+                if (field instanceof BLangRecordLiteral.BLangRecordKeyValueField) {
```

Comment:
You may not these `if` checks as L:123 does the same.

Reply:
I added this second if clause here because of SpotBug error message. As you mentioned since it is redundant to have both the if clauses here, I removed the first one.